api/v1beta1/azurecluster_webhook.go

```diff
@@ -128,6 +128,44 @@ func (c *AzureCluster) ValidateUpdate(oldRaw runtime.Object) error {
 		)
 	}
 
+	oldSubnetMap := make(map[string]SubnetSpec, len(old.Spec.NetworkSpec.Subnets))
```

Comment:
could we extract this subnet immutability validation to its own function?

Reply:
Sure, no problem.